rosenthal/packages/freedesktop.scm

```diff
@@ -46,3 +46,46 @@
                 (sha256
                  (base32
                   "0f72359fzvh6jzri4fd79m34rwm2r55p2ryq4306wrw7xliafzx0")))))))
+
+(define-public xdg-desktop-portal-hyprland
+  (package
+    (name "xdg-desktop-portal-hyprland")
+    (version "0.4.0")
+    (source (origin
+              (method git-fetch)
+              (uri (git-reference
+                     (url "https://github.com/hyprwm/xdg-desktop-portal-hyprland")
+                     (commit (string-append "v" version))))
+              (file-name (git-file-name name version))
+              (sha256
+                (base32
+                  "1dcglmx02j73qbmw9qsacamn8byakyzwknpqjnhsyphixb4crrdg"))))
+    (build-system meson-build-system)
+    (arguments
+      (list
+        #:configure-flags #~(list "-Dsystemd=disabled")))
+    (native-inputs (list cmake pkg-config wayland hyprland-protocols))
+    (inputs (list elogind hyprland-protocols pipewire wayland-protocols wayland libinih `(,util-linux "lib")))
+    (home-page "https://github.com/hyprwm/xdg-desktop-portal-hyprland")
+    (synopsis "@code{xdg-desktop-portal} backend for Hyprland")
+    (description
+      "This package provides @code{xdg-desktop-portal-hyprland}. This project extends @code{xdg-desktop-portal-wlr} for Hyprland, adding support for @code{xdg-desktop-portal} screenshot and casting interfaces while adding a few extra portals specific to Hyprland: mostly for window sharing. Requires @code{hyprland-share-picker}.")
+    (license license:expat)))
+
+(define-public hyprland-share-picker
+  (package
+    (inherit xdg-desktop-portal-hyprland)
+    (name "hyprland-share-picker")
+    (build-system qt-build-system)
+    (inputs (modify-inputs (package-inputs xdg-desktop-portal-hyprland)
+                           (append qtwayland-5 slurp)))
+    (native-inputs (modify-inputs (package-native-inputs xdg-desktop-portal-hyprland)
+                                  (append qtwayland-5)))
```

Comment:
Not needed for native-inputs.

Reply:
Will adjust and refactor my inputs, the whole "native-inputs", "inputs", and "propogated-inputs" still seem to confuse me.

Reply:
See the paragraphs below `propagated-inputs` in [_(guix) package Reference_](<https://guix.gnu.org/en/manual/devel/en/html_node/package-Reference.html>).